main/src/library/MutPriorityQueue.flix

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2023 Xavier deSouza
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+pub enum MutPriorityQueue[a: Type, r: Region] {
+    case MutPriorityQueue(Region[r], Ref[Array[a, r], r], Ref[Int32, r])
+}
+
+mod MutPriorityQueue {
+
+    ///
+    /// Represents a mutable priority queue.
+    /// Explanation of component types (left to right):
+    /// Component 2: A reference to the backing array.
+    /// Component 3: A reference to the number of elements in the mutable priority queue.
+    ///
+    /// The maximum element (if it exists) can always be accessed in constant time.
```

Comment:
Whats with this comment on toString??

Reply:
You asked if it could be done a different way. I said it was the same as the toString functions for other mutable structures.

Reply:
But why does the doc comment for toString say "The maximum element (if it exists) can always be accessed in constant time."

Reply:
Yes, that's my bad. I realize I replaced the comment in the wrong area. Fixing now.

Reply:
The comment was meant for the enum

Reply:
Ok, maybe do pass through the GitHub change to see that everything is in the right place.

Reply:
Ok. It looks correct to me now.